mlx/io/safetensor.cpp

```diff
@@ -0,0 +1,190 @@
+#include "mlx/io/safetensor.h"
+
+#include <stack>
+
+namespace mlx::core {
+
+std::string dtype_to_safetensor_str(Dtype t) {
+  if (t == float32) {
```

Comment:
Could you make this a switch statement, and leave out the default case? The switch statement is kind of handy because if we add a new type the compiler will tell us if we missed adding it some place.